sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -239,22 +239,98 @@ object GpuDivideUtil {
   }
 }
 
+abstract class GpuDecimalDivide(left: Expression, right: Expression) extends GpuDivModLike with
+  Serializable {
+  // Override the output type as a special case for decimal
+  override def dataType: DataType = (left.dataType, right.dataType) match {
+    case (l: DecimalType, r: DecimalType) => GpuDivideUtil.decimalDataType(l, r)
+    case _ => super.dataType
+  }
+
+  def getNewDecimalType(outputScale: Int, l: DecimalType, r: DecimalType): DecimalType = {
+    val newScale = if (outputScale + r.scale > l.scale) {
```

Comment:
FYI: Spark has already make the lhs and rhs precision and scale the same.  This is part of the problem with multiply and divide is that Spark makes them both longer than they need to be so that they can be the same, which makes the spark binary op code happen, but results Spark doing things really inefficiently.